InternxtDesktop/Services/Auth/AuthManager.swift

```diff
@@ -74,6 +74,62 @@ class AuthManager: ObservableObject {
         self.logger.info("Auth details removed correctly, user is logged out")
     }
     
+    func handleSignInDeeplink(url: URL) throws -> Bool {
+        guard url.scheme == "internxt" else {
+            self.logger.info("Invalid scheme")
+            return false
+        }
+        
+        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: true) else {
+            self.logger.info("Invalid URL")
+            return false
+        }
+        
+        guard let action = components.host, action == "login-success" else {
+            self.logger.info("Unknown URL, not handling")
+            return false
+        }
+        
+        guard let base64Mnemonic = components.queryItems?.first(where: { $0.name == "mnemonic" })?.value else {
+            self.logger.info("Mnemonic not found")
+            return false
+        }
+        
+        
+        guard let base64LegacyToken = components.queryItems?.first(where: { $0.name == "token" })?.value else {
+            self.logger.info("Legacy token not found")
+            return false
+        }
+        
+        guard let base64Token = components.queryItems?.first(where: { $0.name == "newToken" })?.value else {
+            self.logger.info("Token not found")
+            return false
+        }
+        
+        guard let decodedToken = Data(base64Encoded: base64Token.data(using: .utf8)!) else {
+            self.logger.info("Cannot decode token")
+            return false
+        }
+        
+        guard let decodedLegacyToken = Data(base64Encoded: base64LegacyToken.data(using: .utf8)!) else {
+            self.logger.info("Cannot decode legacy token")
+            return false
+        }
+        
+        guard let decodedMnemonic = Data(base64Encoded: base64Mnemonic.data(using: .utf8)!) else {
```

Comment:
Guard also that the mnemonic has the proper form, as this is critical and any operation apart from the upload/download (which is late in terms of UX) will tell you that the form is not correct.